hi-ml-histopathology/testhisto/testhisto/mocks/tiles_datamodule.py

```diff
@@ -0,0 +1,165 @@
+    :param MASK_COLUMN: CSV column name for relative path to mask file.
+    :param LABEL_COLUMN: CSV column name for tile label.
+    :param TILE_X_COLUMN: CSV column name for horizontal tile coordinate.
+    :param TILE_Y_COLUMN: CSV column name for vertical tile coordinate.
+    :param DEFAULT_CSV_FILENAME: Default name of the dataset CSV at the dataset rood directory.
+    :param DATA_PROVIDER: CSV column name for data provider .
+    :param METADATA_POSSIBLE_VALUES: Possible values to be assigned to the dataset metadata. The values mapped to
+    isup_grade are the possible gleason_scores.
+    """
+
+    SLIDE_ID_COLUMN = MockTilesDataset.SLIDE_ID_COLUMN
+    TILE_ID_COLUMN = MockTilesDataset.TILE_ID_COLUMN
+    IMAGE_COLUMN = MockTilesDataset.IMAGE_COLUMN
+    MASK_COLUMN = "mask"
+    TILE_X_COLUMN = MockTilesDataset.TILE_X_COLUMN
+    TILE_Y_COLUMN = MockTilesDataset.TILE_Y_COLUMN
+    OCCUPANCY = "occupancy"
+    DEFAULT_CSV_FILENAME = MockTilesDataset.DEFAULT_CSV_FILENAME
+    DATA_PROVIDER = "data_provider"
+    ISUP_GRADE = "slide_isup_grade"
+    GLEASON_SCORE = "gleason_score"
+
+    N_CLASSES = MockTilesDataset.N_CLASSES
+
+    METADATA_POSSIBLE_VALUES: dict = {
+        DATA_PROVIDER: ["site_0", "site_1"],
+        ISUP_GRADE: {
+            0: ["0+0", "negative"],
+            4: ["4+4", "5+3", "3+5"],
+            1: ["3+3"],
+            3: ["4+3"],
+            2: ["3+4"],
+            5: ["4+5", "5+4", "5+5"],
+        },
+    }
+
+    METADATA_COLUMNS = tuple(METADATA_POSSIBLE_VALUES.keys())
+
+    def __init__(self, img_size: int = 224, **kwargs: Any) -> None:
+        self.img_size = img_size
+        super().__init__(**kwargs)
+        assert (
+            self.n_slides >= self.N_CLASSES
+        ), f"The number of slides should be >= self.N_CLASSES (i.e., {self.N_CLASSES})"
+        assert (self.img_size // self.tile_size) ** 2 >= self.n_tiles, (
+            f"The image of size {self.img_size} can't contain more than {(self.img_size // self.tile_size)**2} tiles."
+            "Choose a number of tiles n_tiles <= (img_size // tile_size)**2 "
+        )
+
+    def create_mock_metadata_dataframe(self) -> pd.DataFrame:
+        """Create a mock dataframe with random metadata."""
+        df = pd.DataFrame(
+            columns=[
+                self.SLIDE_ID_COLUMN,
+                self.TILE_ID_COLUMN,
+                self.IMAGE_COLUMN,
+                self.MASK_COLUMN,
+                self.TILE_X_COLUMN,
+                self.TILE_Y_COLUMN,
+                self.OCCUPANCY,
+                self.DATA_PROVIDER,
+                self.ISUP_GRADE,
+                self.GLEASON_SCORE,
+            ]
+        )
+        grid = [
+            f"{i}_{j}" for i in range(self.img_size // self.tile_size) for j in range(self.img_size // self.tile_size)
+        ]
+        coords = list(
+            map(lambda x: list(map(int, x.split("_"))), np.random.choice(grid, size=self.n_tiles, replace=False))
+        )
```

Comment:
because [np.random.choice](https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html) only takes a 1D array as input, it was just a hacky way to sample random distinct 2D coordinates, but I like your way even better! Thanks for the suggestion!